diff --git a/src/runtime/cgo/cgo.go b/src/runtime/cgo/cgo.go
index 1b87fcc8ec..b6cd60798e 100644
--- a/src/runtime/cgo/cgo.go
+++ b/src/runtime/cgo/cgo.go
@@ -14,7 +14,7 @@ package cgo
 #cgo darwin,!arm64 LDFLAGS: -lpthread
 #cgo dragonfly LDFLAGS: -lpthread
 #cgo freebsd LDFLAGS: -lpthread
-#cgo android LDFLAGS: -llog
+//#cgo android LDFLAGS: -llog
 #cgo !android,linux LDFLAGS: -lpthread
 #cgo netbsd LDFLAGS: -lpthread
 #cgo openbsd LDFLAGS: -lpthread
