diff --git a/src/net/conf.go b/src/net/conf.go
index 92c5618d1e..102e720155 100644
--- a/src/net/conf.go
+++ b/src/net/conf.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build !android
+
 package net
 
 import (
diff --git a/src/net/conf_android.go b/src/net/conf_android.go
index 92c5618d1e..45ed6eea10 100644
--- a/src/net/conf_android.go
+++ b/src/net/conf_android.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build android
+
 package net
 
 import (
@@ -420,7 +422,7 @@ func (c *conf) lookupOrder(r *Resolver, hostname string) (ret hostLookupOrder, d
 				var haveMDNSAllow bool
 				switch c.mdnsTest {
 				case mdnsFromSystem:
-					_, err := os.Stat("/etc/mdns.allow")
+					_, err := os.Stat("/data/data/com.termux/files/usr/etc/mdns.allow")
 					if err != nil && !errors.Is(err, fs.ErrNotExist) {
 						// Let libc figure out what is going on.
 						return hostLookupCgo, dnsConf
diff --git a/src/net/dnsclient_android.go b/src/net/dnsclient_android.go
index 6e749882c2..0f62f65194 100644
--- a/src/net/dnsclient_android.go
+++ b/src/net/dnsclient_android.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build android
+
 // DNS client: see RFC 1035.
 // Has to be linked into package net for Dial.
 
@@ -366,7 +368,7 @@ type resolverConfig struct {
 var resolvConf resolverConfig
 
 func getSystemDNSConfig() *dnsConfig {
-	return getSystemDNSConfigNamed("/etc/resolv.conf")
+	return getSystemDNSConfigNamed("/data/data/com.termux/files/usr/etc/resolv.conf")
 }
 
 func getSystemDNSConfigNamed(path string) *dnsConfig {
@@ -378,7 +380,7 @@ func getSystemDNSConfigNamed(path string) *dnsConfig {
 func (conf *resolverConfig) init() {
 	// Set dnsConfig and lastChecked so we don't parse
 	// resolv.conf twice the first time.
-	conf.dnsConfig.Store(dnsReadConfig("/etc/resolv.conf"))
+	conf.dnsConfig.Store(dnsReadConfig("/data/data/com.termux/files/usr/etc/resolv.conf"))
 	conf.lastChecked = time.Now()
 
 	// Prepare ch so that only one update of resolverConfig may
diff --git a/src/net/dnsclient_unix.go b/src/net/dnsclient_unix.go
index 6e749882c2..daddb3addc 100644
--- a/src/net/dnsclient_unix.go
+++ b/src/net/dnsclient_unix.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build !android
+
 // DNS client: see RFC 1035.
 // Has to be linked into package net for Dial.
 
