diff --git a/src/os/export_linux_test.go b/src/os/export_linux_test.go
index 4ace32bb5b..28f31a9a3c 100644
--- a/src/os/export_linux_test.go
+++ b/src/os/export_linux_test.go
@@ -8,7 +8,6 @@ var (
 	PollCopyFileRangeP  = &pollCopyFileRange
 	PollSpliceFile      = &pollSplice
 	GetPollFDAndNetwork = getPollFDAndNetwork
-	CheckPidfdOnce      = checkPidfdOnce
 )
 
 const StatusDone = statusDone
diff --git a/src/os/export_pidfd_test.go b/src/os/export_pidfd_test.go
new file mode 100644
index 0000000000..79f529ca02
--- /dev/null
+++ b/src/os/export_pidfd_test.go
@@ -0,0 +1,9 @@
+// Copyright 2024 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license can be found in the LICENSE file.
+
+//go:build linux && !android
+
+package os
+
+var CheckPidfdOnce = checkPidfdOnce
diff --git a/src/os/pidfd_linux.go b/src/os/pidfd_linux.go
index 796d8c018c..1199a1f621 100644
--- a/src/os/pidfd_linux.go
+++ b/src/os/pidfd_linux.go
@@ -12,6 +12,11 @@
 // N.B. Alternative Linux implementations may not follow this ordering. e.g.,
 // QEMU user mode 7.2 added pidfd_open, but CLONE_PIDFD was not added until
 // 8.0.
+//
+// On Android systems the pidfd_open call reportedly triggers a SIGSYS signal,
+// (issue #69065), so don't use pidfd on Android.
+
+//go:build linux && !android
 
 package os
 
diff --git a/src/os/pidfd_linux_test.go b/src/os/pidfd_linux_test.go
index 0210850c04..e493eff03a 100644
--- a/src/os/pidfd_linux_test.go
+++ b/src/os/pidfd_linux_test.go
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+//go:build linux && !android
+
 package os_test
 
 import (
diff --git a/src/os/pidfd_other.go b/src/os/pidfd_other.go
index 7a282307db..df72ebbab9 100644
--- a/src/os/pidfd_other.go
+++ b/src/os/pidfd_other.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build (unix && !linux) || (js && wasm) || wasip1 || windows
+//go:build (unix && (android || !linux)) || (js && wasm) || wasip1 || windows
 
 package os
 
