diff --git a/go-src/src/runtime/plugin.go b/go-src/src/runtime/plugin.go
index 40dfefde17..eb933a8313 100644
--- a/go-src/src/runtime/plugin.go
+++ b/go-src/src/runtime/plugin.go
@@ -48,12 +48,12 @@ func plugin_lastmoduleinit() (path string, syms map[string]any, initTasks []*ini
 			throw("plugin: new module data overlaps with previous moduledata")
 		}
 	}
-	for _, pkghash := range md.pkghashes {
-		if pkghash.linktimehash != *pkghash.runtimehash {
-			md.bad = true
-			return "", nil, nil, "plugin was built with a different version of package " + pkghash.modulename
-		}
-	}
+	// for _, pkghash := range md.pkghashes {
+	// 	if pkghash.linktimehash != *pkghash.runtimehash {
+	// 		md.bad = true
+	// 		return "", nil, nil, "plugin was built with a different version of package " + pkghash.modulename
+	// 	}
+	// }
 
 	// Initialize the freshly loaded module.
 	modulesinit()
